### ISSUE TITLE
How to re-open Emacs in Gui

### ISSUE THREAD

**Issue by krayF1sh** [hubisan/emacs-wsl#47]
Hello and thank you for this guide, it is really helpful. 
One thing I can't make is, if I need to restart Emacs, what command should I run in terminal in order to re-open it in GUI? 

`setsid emacs` gives me an error "emacs: Could not open file: /dev/tty" and I need to close everything and run the "wsl-2_0-run-all.bat" again. 

Is there any easier way to do this?

**Comment by hubisan** [comment]
You need to set the DISPLAY beforehand by running the following commands (the second line is for direct rendering):

```
export DISPLAY=$(ip route | awk '/^default/{print $3; exit}'):0.0
export LIBGL_ALWAYS_INDIRECT=1
setsid emacs
 ```

You can add an alias as described in the [README](https://github.com/hubisan/emacs-wsl/blob/master/README.org#run-emacs) to run the above commands with just `ema` or whatever alias you wish.

**Comment by krayF1sh** [comment]
I am might doing something wrong because I tried it and it doesn't work. I reached out to you after tried everything. 

This is what I am getting:
```
exit}):0.0
export LIBGL_ALWAYS_INDIRECT=1
setsid emacs
: command not found
```

**Comment by hubisan** [comment]
- Did you run Emacs with the `wsl-2_0-run-all.bat` or directly from the terminal the first time?
- What do you mean by restarting? Did you already kill Emacs and you want to launch it again or is Emacs still running?
- What happens if you launch `wsl-2_0-run-all.bat` again to restart Emacs? Does this work?
- What happens if you just run `emacs` to restart it (without the setsid)? Does it launch Emacs in GUI and keeps the terminal open? Or do you get an error as well?
- What happens if you enter the following line in the terminal (I hope it is correct, can't test it atm)?
  ``` shell
  export DISPLAY=$(ip route | awk '/^default/{print $3; exit}'):0.0 && export LIBGL_ALWAYS_INDIRECT=1 && setsid emacs
  
  # Another version I found on my computer, maybe the first one is incorrect:
  export DISPLAY=$(ip route | awk '/default via / {print $3; exit}' 2>/dev/null):0.0 && export LIBGL_ALWAYS_INDIRECT=1 && setsid emacs
  ```
- If both fail, restart the terminal and try it again. Does this work?
- Or try `unset DISPLAY` before running the command?
- Or mabye change `setsid emacs` to `sudo setsid emacs`. This is mentionned in the [emacswiki](https://www.emacswiki.org/emacs/Emacs_and_the_Windows_Subsystem_for_Linux#h5o-5).

### My usual workflow

1. Run Firewall rule and VcXsrv with the script `./run-emacs-wsl-2/wsl-2_1-firewall-rule-vcxsrv.bat`
2. Open terminal.
3. Run emacs with the alias `ema` as described in the README. If have added `exit` to my alias to kill the terminal. Maybe that is why I don't have any problems. I launch Emacs from the terminal because lsp-mode doesn't seem to work if I use the script.
4. If I want to restart Emacs:
    - <kbd>M-x kill-emacs</kbd> or <kbd>C-x C-c</kbd>
    - Launch a new terminal.
    - Run that alias again.

**Comment by krayF1sh** [comment]
Hey there, thank you for your reply.

1. I don't remember. I think I ran `emacs -nw`
2. Yes, atm when I kill Emacs I need to close everything and run `wsl-2_0-run-all.bat`
3. Yep, but I need to close VcXsrv first
4. just emacs opens emacs in terminal
5. Both opened emacs in GUI. In addition, after killed Emacs and used them Emacs opened just fine

I believe something is wrong with the alias part. I paste a screenshot 
![Καταγραφή](https://user-images.githubusercontent.com/42109143/128222910-b88946c5-47d0-4de0-a94a-d52bbfa1215f.PNG)
The part `/default via / {print $3; exit}` is not in green and if I add this whole thing to ~/.bashrc and restart the terminal then I am getting the above error every time. But the commands are running fine.

P.S. I have used all the combinations for the alias part, same situation when adding 
```
alias ema='
export DISPLAY=$(ip route | awk '/^default/{print $3; exit}'):0.0
export LIBGL_ALWAYS_INDIRECT=1
setsid emacs
'
```

**Comment by hubisan** [comment]
Try the following, changed the surrounding from `'` to `"`:

``` shell
alias ema="
export DISPLAY=$(ip route | awk '/^default/{print $3; exit}'):0.0
export LIBGL_ALWAYS_INDIRECT=1
setsid emacs
"
```

Forgot to change that in the README. Sorry for the inconvenience.

**Comment by krayF1sh** [comment]
Cool, that worked! Thank you.

**Comment by hubisan** [comment]
Fixed with #49
[v1.1.3](https://github.com/hubisan/emacs-wsl/compare/v1.1.2...v1.1.3)
